flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/HttpRequestHandler.java

```diff
@@ -99,7 +110,43 @@ public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) {
 					currentDecoder.destroy();
 					currentDecoder = null;
 				}
-				
+
+				if(secureCookie != null) {
```

Comment:
Why do you decide here if security is enabled based on the COOKIE's existence. I think it could happen that users torn security off, but still have some cookie set.